client/my-sites/email/email-providers-stacked-comparison/provider-cards/utils.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+	RequestCartProduct,
+	ResponseCart,
+	ShoppingCartManagerActions,
+} from '@automattic/shopping-cart';
+import page from 'page';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+import { IncompleteRequestCartProduct } from 'calypso/lib/cart-values/cart-items';
+
+export enum IntervalLength {
+	ANNUALLY = 'annually',
+	MONTHLY = 'monthly',
+}
+
+export const addToCartAndCheckout = (
+	shoppingCartManager: ShoppingCartManagerActions,
+	cartItem: RequestCartProduct | IncompleteRequestCartProduct,
+	productList: Record< string, { product_id: number } >,
+	setAddingToCart: ( addingToCart: boolean ) => void,
+	selectedSite: string
+): void => {
+	setAddingToCart( true );
+
+	shoppingCartManager
+		.addProductsToCart( [ fillInSingleCartItemAttributes( cartItem, productList ) ] )
+		.then( ( response: ResponseCart ) => {
+			setAddingToCart( false );
+
+			const errors = response?.messages?.errors;
+
+			if ( errors && errors.length ) {
+				// Stay on the page to show the relevant error(s)
+				return;
+			}
+
+			page( `/checkout/${ selectedSite }` );
+		} )
+		.catch( () => {
+			setAddingToCart( false );
+		} );
+};
+
+export const recordTracksEventAddToCartClick = (
+	comparisonContext: string,
+	validatedMailboxUuids: string[],
+	mailboxesAreValid: boolean,
+	provider: string,
+	source: string,
+	userCanAddEmail: boolean,
+	userCannotAddEmailReason?: { code: string } | null
+): void => {
```

Comment:
Nit: I think it would be cleaner and more future-proof to have these supplied as properties of single object rather than individual arguments.
```suggestion
export const recordTracksEventAddToCartClick = ( {
	comparisonContext: string,
	validatedMailboxUuids: string[],
	mailboxesAreValid: boolean,
	provider: string,
	source: string,
	userCanAddEmail: boolean,
	userCannotAddEmailReason?: { code: string } | null
} ): void => {
```